extensions/blocks/story/edit.native.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Text, View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { Image } from '@wordpress/components';
+import { BlockMediaUpdateProgress } from '@wordpress/block-editor';
+import { __, sprintf } from '@wordpress/i18n';
+import { getProtocol } from '@wordpress/url';
+import { doAction, hasAction } from '@wordpress/hooks';
+import {
+	// TODO implement the similar / following bridge signals
+	// requestImageFailedRetryDialog,
+	// requestImageUploadCancelDialog,
+	// requestImageFullscreenPreview,
+	mediaUploadSync,
+	mediaSaveSync,
+	requestMediaFilesEditorLoad,
+} from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import StoryEditingButton from './story-editing-button';
+
+class StoryEdit extends React.Component {
```

Comment:
oh thanks for the pointers @jeherve ! I'm a bit rusty since I switched back to work on mainly native side of the apps for over a year now 😅 - @jd-alexander made a [similar observation here](https://github.com/Automattic/jetpack/pull/17222/files#r503574596), I didn't realize it would need to be enforced, and besides me not being up to date with latest practices in R/RN I thought it was a safer bet to follow along how other components that are already in place work (for example Image or Video blocks, check the VideoEdit or ImageEdit classes in their native.js counterparts), so to make it easier. Will read, learn and look into changing it 👍

Reply:
PR implementing function component ready here https://github.com/Automattic/jetpack/pull/17549 👍